lib/DownloadUtils.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Created by AsherS on 8/20/15.
+ */
+
+var fs = require('fs');
+var request = require('request');
+var Q = require('q');
+
+var DownloadUtils = (function(){
+
+    function DownloadUtils(){
+
+    }
+
+    DownloadUtils.downloadFile = function (url,dest) {
+
+        var deferred = Q.defer();
+
+        try {
+            var req = request.get(url);
+            req
```

Comment:
What's request max connection value? If it uses http internally, than the Node.js http client automatically uses socket pooling: by default, this limits you to 5 sockets per host which is quite low for downloading ts files from the same Wowza host.  Consider enlarging it. See http://nodejs.org/docs/v0.6.3/api/http.html#agent.maxSockets